Improve logging and exception handling

RagnarW opened the pull request:
Ensure we always re-throw exception when server fails to bind.

Add logging when server has bound.

Add logging when reconnecting channel fails to connect.

Failed request during store copy are now info and not error since they are expected to happen every now and then.

RagnarW committed:
More logging

Added log when server is bound to channel

Added log when ReconnectedChannel fails to connect.

RagnarW committed:
Make sure exception is always rethrown when Server fails to start

martinfurmanski commented on enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/ReconnectingChannel.java:
```diff
@@ -95,7 +94,9 @@ else if ( fChannel != null && !fChannel.isDone() )
             {
-                f.channel().eventLoop().schedule( this::tryConnect, connectionBackoff.getMillis(), MILLISECONDS );
+                long millis = connectionBackoff.getMillis();
+                log.warn( "Failed to connect to: %s. Retrying in %d ms", destination.socketAddress(), millis );
```
This will spam like crazy. Consider a CappedLogger with an interval of at least 10 seconds.

RagnarW committed:
Use capped logger for connection issue warnings

RagnarW committed:
Improve request logging during store copy

Now logs warn instead of error on failure. Some failed requests are expected.

Provide more information about the cause of failure.

martinfurmanski commented on enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreCopyClient.java:
```diff
@@ -110,20 +110,24 @@ private void persistentCall( CatchUpRequest request, CatchUpResponseAdaptor<Stor
                 log.info( format( "Sending request '%s' to '%s'", request, from ) );
                 StoreCopyFinishedResponse response = catchUpClient.makeBlockingRequest( from, request, copyHandler );
                 successful = successfulFileDownload( response );
```
You already log in successfulFileDownload. Can't you update the logging logic there?

RagnarW commented:
@martinfurmanski Moved the logging
